### PR TITLE
RFC: avocado: Replace custom skip* decorators with unittests ones

### PR DESCRIPTION
**Pull request by ldoktor**
Instead of re-implementing the wheel let's just inherit unittest's
decorators and support their privileged usage (exceptions inside test's
run) in the avocado tests. The fact that we inherit doesn't mean we
should promote this. We should still encourage people to use
`avocado.skip*` methods because at some point we might want to
improve/modify/update the decorators which could be again new
implementation or just some enhancement of the inherited ones.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

File: selftests/functional/test_skiptests.py [hunk 1]
```diff
@@ -32,15 +33,16 @@ def test2(self):
     def test3(self):
         pass
 
-    @avocado.skipIf(False)
+    @avocado.skipIf(False, "won't be skipped")
     def test4(self):
         pass
 
-    @avocado.skipUnless(True)
+    @avocado.skipUnless(True, "won't be skipped")
```
Comment (lmr) [inline selftests/functional/test_skiptests.py]: Are you sure this is supposed to not be executed? I got the exact opposite impression

https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures

So the comment message should be 'will be skipped' instead.

File: avocado/core/decorators.py [hunk 2]
```diff
@@ -53,39 +53,3 @@ def wrap(*args, **kwargs):
...
-    """
-    Decorator to skip a test if a condition is False.
-    """
-    if not condition:
-        return skip(message)
-    return _itself
-
-
-def _itself(obj):
-    return obj
```
Comment (apahim) [inline avocado/core/decorators.py]: They are not really the same. The code is very close indeed, but the unittest decorators are setting attributes in the test method to communicate the reason of the skip. We have a different architecture. That's the main reason I decided to implement our own decorators.

File: selftests/functional/test_skiptests.py (hunk 1 above)
Comment (ldoktor) [inline selftests/functional/test_skiptests.py]: Oups, my fault. Sorry I did this in a hurry just to demonstrate this version... I'll wait for the response and then I'll polish it (if accepted)

File: selftests/functional/test_skiptests.py [hunk 3]
```diff
@@ -32,15 +33,16 @@ def test2(self):
...
     def test5(self):
         pass
 
-    @avocado.skip()
+    @unittest.skipIf(True, "Skipped using unittest's skipIf which works"
+                     "but is not recommended.")
```
Comment (apahim) [inline selftests/functional/test_skiptests.py]: The fact that this works is a limiting factor for future improvements in our skip decorators.

Comment (ldoktor) [inline selftests/functional/test_skiptests.py]: The point is that (as mentioned in the docstring) we don't guarantee this to work and we still encourage people to use `avocado.skip*` methods.

File: avocado/core/decorators.py (hunk 2 above)
Comment (ldoktor) [inline avocado/core/decorators.py]: Well my point is that I'd prefer re-using as much from `unittest` as possible, because it's proven to work well and users expect it to behave that way. For this reason I'd prefer us to try to adjust avocado in order to be more-like unittest (this is just a draft to start the discussion, not a polished version).

File: selftests/functional/test_skiptests.py (hunk 3 above)
Comment (apahim) [inline selftests/functional/test_skiptests.py]: Yep, both your point and the docstring are very clear. I'm sharing my view on this, as requested.

Comment (lmr) [inline selftests/functional/test_skiptests.py]: We can simply say it's not supported at all

**Comment by adereis** [comment]
I don't see advantages in reusing the `unittests` API here, **especially** if we actually don't want users to use the native APIs as you describe. And I don't think supporting just this particular decorator from the `unittest` module is a good idea given other decorators from the same module are not supported. It brings confusion to the API documentation and would force us to support it forever.

In summary, we would save a few lines of code at the cost of extra confusion, need of extra documentation and/or API stiffness. Not a good trade-off.

**Comment by ldoktor** [comment]
OK, I'm just lazy so I always want to simplify my work by relying on others code. Anyway as I mentioned I understand this is not a clean solution, my idea was that the skip decorators for `test` are against avocado's decisions so I wanted to simply allow this from unittest and say it's their thing, we just support it for convenience reasons, but I see it's not popular so let's close this one.
